var/spack/repos/builtin/packages/qt/package.py

```diff
@@ -95,8 +95,19 @@ def url_for_version(self, version):
 
         return "{0}/{1}".format(Qt.base_url, url_ext.format(url_vstr, version))
 
-    def setup_environment(self, spack_env, env):
-        env.set('QTDIR', self.prefix)
+    def setup_environment(self, spack_env, run_env):
+        run_env.set('QTDIR', self.prefix)
+
+        # NOTE: The Qt package links against an internal library named 'pcre',
+        # which causes build problems for versions of Qt that depend on the
+        # Spack library with the same name.  To avoid these build issues,
+        # affected Qt installs remove the Spack 'pcre' from their path.
+        # TODO: This is a pretty hacky way of solving this problem that should
+        # be revised to ignore Spack 'pcre' instead of changing package deps.
+        qt_deps = self.spec.traverse(root=False, deptype='build')
+        qt_pcre_dep = next((d for d in qt_deps if d.name == 'pcre'), None)
+        if qt_pcre_dep:
+            spack_env.remove_path('SPACK_DEPENDENCIES', qt_pcre_dep.prefix)
```

Comment:
I think that prefixing qt would be an improved solution for sure as it would account for any other Spack library conflicts that may arise.  I'll see what I can do.